src/mlpack/bindings/cli/add_to_po.hpp

```diff
@@ -2,110 +2,101 @@
 
+#include <mlpack/third_party/CLI/CLI11.hpp>
+
 namespace mlpack {
 namespace bindings {
-namespace cli {
+namespace cmd {
 
 /**
- * Add a non-vector option to boost::program_options.
- *
- * @param boostName The name of the option to add to boost::program_options.
- * @param descr Description string for parameter.
- * @param desc Options description to add parameter to.
- */
-template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
-             const typename boost::disable_if<util::IsStdVector<T>>::type* = 0,
-             const typename boost::disable_if<std::is_same<T, bool>>::type* = 0)
-{
-  desc.add_options()(boostName.c_str(), boost::program_options::value<T>(),
-      descr.c_str());
-}
-
-/**
- * Add a vector option to boost::program_options.  This overload will use the
- * multitoken() option.
- *
- * @param boostName The name of the option to add to boost::program_options.
+ * Add a vector option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
  * @param descr Description string for parameter.
  * @param desc Options description to add parameter to.
  */
 template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
              const typename boost::enable_if<util::IsStdVector<T>>::type* = 0,
              const typename boost::disable_if<std::is_same<T, bool>>::type* = 0)
 {
-  desc.add_options()(boostName.c_str(),
-      boost::program_options::value<T>()->multitoken(), descr.c_str());
+  app.add_option_function<T>(cliName.c_str(), 
+  [&param](const T& value)
+  {
+     param.value = value;
+     param.wasPassed = true;
+  }, 
+  param.desc.c_str());
 }
 
 /**
- * Add a boolean option to boost::program_options.
+ * Add a boolean option to CLI11.
  *
- * @param boostName The name of the option to add to boost::program_options.
+ * @param cliName The name of the option to add to CLI11.
  * @param descr Description string for parameter.
  * @param desc Options description to add parameter to.
  */
 template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
              const typename boost::disable_if<util::IsStdVector<T>>::type* = 0,
              const typename boost::enable_if<std::is_same<T, bool>>::type* = 0)
 {
-  desc.add_options()(boostName.c_str(), descr.c_str());
+  app.add_flag_function(cliName.c_str(), 
+  [&param](const T& value)
+  {
+    param.value = value;
+    param.wasPassed = true;
+  }, 
+  param.desc.c_str());
```

Comment:
> @rcurtin if these two can be merged, there would be no need for SFINAE since there will be no failure in substitution, what do you think?

I can't seem to comment on that thread while writing a new review, but I agree; perhaps we can combine everything into the bottom `AddToPO(util::ParamData&, const void*, void*)` function for simplicity. :+1: